src/Juvix/Backends/Michelson/Compilation/Prim.hs

```diff
@@ -28,34 +28,35 @@ primToInstr prim ty =
   --        operations instead of just promoting them
   -- If arguments are both constants, drop both from vstack, PUSH the addition (e.g.)
   -- If one is real, one is not, push the constant to the stack, add an ADD instruction.
-  case prim of
-    -- :: \x -> y ~ s => (f, s)
-    Env.PrimFst → do
-      let ErasedAnn.Pi _ (ErasedAnn.PrimTy (Env.PrimTy (M.Type (M.TPair _ _ xT _) _))) _ = ty
-      promoted ← promoteInStack 1
-      modify @"stack" (VStack.cons (VStack.Val VStack.FuncResultE, xT) . VStack.drop 1)
-      pure (M.SeqEx (promoted <> [M.PrimEx (M.CAR "" "")]))
-    Env.PrimSnd → do
-      let ErasedAnn.Pi _ (ErasedAnn.PrimTy (Env.PrimTy (M.Type (M.TPair _ _ _ yT) _))) _ = ty
-      promoted ← promoteInStack 1
-      modify @"stack" (VStack.cons (VStack.Val VStack.FuncResultE, yT) . VStack.drop 1)
-      pure (M.SeqEx (promoted <> [M.PrimEx (M.CDR "" ""), M.PrimEx (M.CAR "" "")]))
-    Env.PrimPair → do
-      let ErasedAnn.Pi _ firstArgTy (ErasedAnn.Pi _ secondArgTy _) = ty
-      xT ← Type.typeToType firstArgTy
-      yT ← Type.typeToType secondArgTy
-      let pairTy = M.Type (M.TPair "" "" xT yT) ""
-      promoted ← promoteInStack 2
-      modify @"stack" (VStack.cons (VStack.Val VStack.FuncResultE, pairTy) . VStack.drop 2)
-      pure (M.SeqEx (promoted <> [M.PrimEx (M.PAIR "" "" "" "")]))
-    -- :: a ~ s => (a, s)
-    Env.PrimConst const → do
-      case const of
-        M.ValueNil → do
-          let ErasedAnn.PrimTy (Env.PrimTy t@(M.Type (M.TList elemTy) _)) = ty
-          modify @"stack" (VStack.cons (VStack.Val VStack.FuncResultE, t))
-          pure (M.PrimEx (M.NIL "" "" elemTy))
-        _ → do
-          let ErasedAnn.PrimTy (Env.PrimTy t) = ty
-          modify @"stack" (VStack.cons (VStack.Val VStack.FuncResultE, t))
-          pure (M.PrimEx (M.PUSH "" t const))
+  undefined
```

Comment:
Still waiting on modifying this function?

Reply:
This code will likely just be erased, once everything is setup